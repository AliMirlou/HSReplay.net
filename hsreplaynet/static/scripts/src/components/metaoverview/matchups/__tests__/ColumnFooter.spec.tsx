import React from "react";
import ColumnFooter from "../ColumnFooter";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

describe("ColumnFooter", () => {
	test("renders correctly", () => {
		const onWeightChanged = jest.fn();
		const component = renderer.create(
			<ColumnFooter
				weight={25}
				onWeightChanged={onWeightChanged}
				max={100}
			/>,
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
		expect(onWeightChanged).not.toHaveBeenCalled();
	});

	test("renders correctly when editable", () => {
		const onWeightChanged = jest.fn();
		const component = renderer.create(
			<ColumnFooter
				weight={25}
				editable
				onWeightChanged={onWeightChanged}
				max={100}
			/>,
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
		expect(onWeightChanged).not.toHaveBeenCalled();
	});

	test("allows inputting a custom weight", () => {
		const onWeightChanged = jest.fn();
		const wrapper = mount(
			<ColumnFooter
				weight={25}
				editable
				onWeightChanged={onWeightChanged}
				max={100}
			/>,
		);
		expect(wrapper).toMatchSnapshot("25");
		expect(onWeightChanged).not.toHaveBeenCalled();
		(wrapper.find("input").instance() as any).value = "2";
		wrapper.find("input").simulate("change");
		expect(wrapper).toMatchSnapshot("2");
		expect(onWeightChanged).toHaveBeenCalledWith(2);
	});
});
