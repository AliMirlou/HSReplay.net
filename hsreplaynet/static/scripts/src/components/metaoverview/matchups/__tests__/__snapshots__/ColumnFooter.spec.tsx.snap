// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnFooter allows inputting a custom weight: 2 1`] = `
<ColumnFooter
  editable={true}
  max={100}
  onWeightChanged={
    [MockFunction] {
      "calls": Array [
        Array [
          2,
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
  weight={25}
>
  <div
    className="matchup-column-footer"
  >
    <Bar
      direction={0}
      label="25.00%"
      total={100}
      value={25}
      valueElement={
        <input
          className="input-popularity"
          min={0}
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
          value="2"
        />
      }
    >
      <div
        className="bar bar-vertical"
      >
        <div
          className="bar-body"
        >
          <div
            style={
              Object {
                "height": "25%",
              }
            }
          />
        </div>
        <input
          className="input-popularity"
          min={0}
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
          value="2"
        />
      </div>
    </Bar>
  </div>
</ColumnFooter>
`;

exports[`ColumnFooter allows inputting a custom weight: 25 1`] = `
<ColumnFooter
  editable={true}
  max={100}
  onWeightChanged={[MockFunction]}
  weight={25}
>
  <div
    className="matchup-column-footer"
  >
    <Bar
      direction={0}
      label="25.00%"
      total={100}
      value={25}
      valueElement={
        <input
          className="input-popularity"
          min={0}
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
          value="25"
        />
      }
    >
      <div
        className="bar bar-vertical"
      >
        <div
          className="bar-body"
        >
          <div
            style={
              Object {
                "height": "25%",
              }
            }
          />
        </div>
        <input
          className="input-popularity"
          min={0}
          onBlur={[Function]}
          onChange={[Function]}
          type="number"
          value="25"
        />
      </div>
    </Bar>
  </div>
</ColumnFooter>
`;

exports[`ColumnFooter renders correctly 1`] = `
<div
  className="matchup-column-footer"
>
  <div
    className="bar bar-vertical"
  >
    <div
      className="bar-body"
    >
      <div
        style={
          Object {
            "height": "25%",
          }
        }
      />
    </div>
    <div
      className="bar-label"
    >
      25.00%
    </div>
  </div>
</div>
`;

exports[`ColumnFooter renders correctly when editable 1`] = `
<div
  className="matchup-column-footer"
>
  <div
    className="bar bar-vertical"
  >
    <div
      className="bar-body"
    >
      <div
        style={
          Object {
            "height": "25%",
          }
        }
      />
    </div>
    <input
      className="input-popularity"
      min={0}
      onBlur={[Function]}
      onChange={[Function]}
      type="number"
      value="25"
    />
  </div>
</div>
`;
